templates/external_cloud_setup.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+
+if [ "$#" -ne 1 ]; then
+    echo "One param for cluster-name needed"
+    echo "The usage: external_cloud_setup.sh <cluster-name>"
+    exit 1
+fi
+
+if ! command -v clusterctl &> /dev/null
+then
+    echo "clusterctl could not be found, make sure you download it and put into right place"
+    exit 1
+fi
+
+if ! command -v ./create_cloud_conf.sh &> /dev/null
+then
+    echo "./create_cloud_conf.sh could not be found, make sure you run this script inside desired folder"
+    exit 1
+fi
+
+
+
+CLUSTER_NAME=$1
+
+echo "##########################################"
+echo "Create kubeconfig file ${CLUSTER_NAME}.kubeconfig for cluster ${CLUSTER_NAME}"
+clusterctl get kubeconfig ${CLUSTER_NAME} --namespace default > ./${CLUSTER_NAME}.kubeconfig
+
+echo "##########################################"
+echo "Create secret cloud-config"
+./create_cloud_conf.sh clouds.yaml openstack > /tmp/cloud.conf
```

Comment:
What do you think about using `mktemp -d` for creating a temporary directory where the `cloud.conf` can be saved to? And afterwards you can remove that directory.

Reply:
yes , will do that